src/js/components/deployments/deployment-wizard/softwaredevices.js

```diff
@@ -218,16 +218,22 @@ export const Software = ({ commonClasses, deploymentObject, releaseRef, releases
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [device, releases]);
 
-  const onReleaseSelectionChange = release => {
-    if (release !== deploymentObject.release) {
-      setDeploymentSettings({ release });
-    }
-  };
+  const onReleaseSelectionChange = useCallback(
+    release => {
+      if (release !== deploymentObject.release) {
+        setDeploymentSettings({ release });
+      }
+    },
+    [deploymentObject.release, setDeploymentSettings]
+  );
 
-  const onReleaseInputChange = inputValue => {
-    setIsLoadingReleases(!releases.length);
-    return dispatch(getReleases({ page: 1, perPage: 100, searchTerm: inputValue, searchOnly: true })).finally(() => setIsLoadingReleases(false));
-  };
+  const onReleaseInputChange = useCallback(
+    inputValue => {
+      setIsLoadingReleases(!releases.length);
+      return dispatch(getReleases({ page: 1, perPage: 100, searchTerm: inputValue, searchOnly: true })).finally(() => setIsLoadingReleases(false));
+    },
+    [dispatch, releases.length]
```

Comment:
Only for my understanding: why dispatch is listed as dependency?

Reply:
Because it gets set through the useDispatch hook above (we now have linting rules for this 👌), so it might potentially change over the component lifetime...